diff --git a/Makefile b/Makefile
index cd3bb79..ac1bb25 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ define Package/luci-app-adguardhome
 	TITLE:=LuCI app for adguardhome
 	PKG_MAINTAINER:=<https://github.com/rufengsuixing/luci-app-adguardhome>
 	PKGARCH:=all
-	DEPENDS:=+!wget&&!curl:wget
+	DEPENDS:=+bash +curl +ca-bundle +xz-utils
 endef
 
 define Package/luci-app-adguardhome/description
diff --git a/root/usr/share/AdGuardHome/update_core.sh b/root/usr/share/AdGuardHome/update_core.sh
index 9450207..4f92454 100755
--- a/root/usr/share/AdGuardHome/update_core.sh
+++ b/root/usr/share/AdGuardHome/update_core.sh
@@ -14,11 +14,14 @@ check_if_already_running(){
 }
 
 check_wgetcurl(){
-	which curl && downloader="curl -L -k --retry 2 --connect-timeout 20 -o" && return
-	which wget-ssl && downloader="wget-ssl --no-check-certificate -t 2 -T 20 -O" && return
-	[ -z "$1" ] && opkg update || (echo error opkg && EXIT 1)
-	[ -z "$1" ] && (opkg remove wget wget-nossl --force-depends ; opkg install wget ; check_wgetcurl 1 ;return)
-	[ "$1" == "1" ] && (opkg install curl ; check_wgetcurl 2 ; return)
+	if command -v curl >/dev/null 2>&1; then
+		downloader="curl -L -k --retry 2 --connect-timeout 20 -o"
+		return
+	fi
+	if command -v wget >/dev/null 2>&1; then
+		downloader="wget --no-check-certificate -t 2 -T 20 -O"
+		return
+	fi
 	echo error curl and wget && EXIT 1
 }
 check_latest_version(){
@@ -54,7 +56,7 @@ check_latest_version(){
 	fi
 }
 doupx(){
-	Archt="$(opkg info kernel | grep Architecture | awk -F "[ _]" '{print($2)}')"
+	Archt="$(uname -m)"
 	case $Archt in
 	"i386")
 	Arch="i386"
@@ -65,6 +67,9 @@ doupx(){
 	;;
 	"x86")
 	Arch="amd64"
+	;;
+	"x86_64")
+	Arch="amd64"
 	;;
 	"mipsel")
 	Arch="mipsel"
@@ -105,7 +110,7 @@ doupx(){
 	upx_latest_ver="$($downloader - https://api.github.com/repos/upx/upx/releases/latest 2>/dev/null|grep -E 'tag_name' |grep -E '[0-9.]+' -o 2>/dev/null)"
 	$downloader /tmp/upx-${upx_latest_ver}-${Arch}_linux.tar.xz "https://github.com/upx/upx/releases/download/v${upx_latest_ver}/upx-${upx_latest_ver}-${Arch}_linux.tar.xz" 2>&1
 	#tar xvJf
-	which xz || (opkg list | grep ^xz || opkg update && opkg install xz) || (echo "xz download fail" && EXIT 1)
+	command -v xz >/dev/null 2>&1 || (echo "xz is required" && EXIT 1)
 	mkdir -p /tmp/upx-${upx_latest_ver}-${Arch}_linux
 	xz -d -c /tmp/upx-${upx_latest_ver}-${Arch}_linux.tar.xz| tar -x -C "/tmp" >/dev/null 2>&1
 	if [ ! -e "/tmp/upx-${upx_latest_ver}-${Arch}_linux/upx" ]; then
@@ -118,7 +123,7 @@ doupdate_core(){
 	echo -e "Updating core..." 
 	mkdir -p "/tmp/AdGuardHomeupdate"
 	rm -rf /tmp/AdGuardHomeupdate/* >/dev/null 2>&1
-	Archt="$(opkg info kernel | grep Architecture | awk -F "[ _]" '{print($2)}')"
+	Archt="$(uname -m)"
 	case $Archt in
 	"i386")
 	Arch="386"
@@ -128,6 +133,9 @@ doupdate_core(){
 	;;
 	"x86")
 	Arch="amd64"
+	;;
+	"x86_64")
+	Arch="amd64"
 	;;
 	"mipsel")
 	Arch="mipsle"
diff --git a/root/usr/share/AdGuardHome/gfw2adg.sh b/root/usr/share/AdGuardHome/gfw2adg.sh
index b32a133..3cc35d0 100755
--- a/root/usr/share/AdGuardHome/gfw2adg.sh
+++ b/root/usr/share/AdGuardHome/gfw2adg.sh
@@ -20,7 +20,8 @@ if [ ! -f "$configpath" ]; then
 	echo "please make a config first"
 	exit 1
 fi
-wget-ssl --no-check-certificate https://cdn.jsdelivr.net/gh/gfwlist/gfwlist/gfwlist.txt -O- | base64 -d > /tmp/gfwlist.txt
+curl -fL --retry 2 --connect-timeout 20 \
+	https://cdn.jsdelivr.net/gh/gfwlist/gfwlist/gfwlist.txt | base64 -d > /tmp/gfwlist.txt
 cat /tmp/gfwlist.txt | awk -v upst="$gfwupstream" 'BEGIN{getline;}{
 s1=substr($0,1,1);
 if (s1=="!")
